exporter/azuredataexplorerexporter/adx_exporter.go

```diff
@@ -0,0 +1,281 @@
+		return err
+	}
+	// Since the transform succeeded ,  using the option for ingestion ingest the data into ADX
+	for _, tm := range transformedadxmetrics {
+		adxmetricjsonstring, err := jsoniter.MarshalToString(tm)
+		if err != nil {
+			e.logger.Error("Error performing serialization of data.", zap.Error(err))
+		}
+		metricsBuffer = append(metricsBuffer, adxmetricjsonstring)
+
+	}
+	if len(metricsBuffer) != 0 {
+		if err := e.ingestData(metricsBuffer); err != nil {
+			return err
+		}
+	}
+	metricsflushed := len(transformedadxmetrics)
+	e.logger.Sugar().Infof("Flushing %d metrics to sink", metricsflushed)
+	return nil
+}
+
+func (e *adxDataProducer) ingestData(b []string) error {
+
+	ingestreader := strings.NewReader(strings.Join(b, nextline))
+
+	if _, err := e.ingestor.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
+		e.logger.Error("Error performing managed data ingestion.", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+func (e *adxDataProducer) logsDataPusher(ctx context.Context, logData plog.Logs) error {
+	resourcelogs := logData.ResourceLogs()
+	var logsBuffer []string
+	for i := 0; i < resourcelogs.Len(); i++ {
+		resource := resourcelogs.At(i)
+		scopelogs := resourcelogs.At(i).ScopeLogs()
+		for j := 0; j < scopelogs.Len(); j++ {
+			scope := scopelogs.At(j)
+			logs := scopelogs.At(j).LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				logdata := logs.At(k)
+				transformedadxlog := mapToAdxLog(resource.Resource(), scope.Scope(), logdata, e.logger)
+				adxlogjsonbytes, err := jsoniter.Marshal(transformedadxlog)
+				if err != nil {
+					e.logger.Error("Error performing serialization of data.", zap.Error(err))
+				}
+				logsBuffer = append(logsBuffer, string(adxlogjsonbytes))
+			}
+		}
+	}
+	// Takes care of the residual data.
+	if len(logsBuffer) != 0 {
+		if err := e.ingestData(logsBuffer); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *adxDataProducer) tracesDataPusher(ctx context.Context, traceData ptrace.Traces) error {
+	resourcespans := traceData.ResourceSpans()
+	var spanBuffer []string
+	for i := 0; i < resourcespans.Len(); i++ {
+		resource := resourcespans.At(i)
+		scopespans := resourcespans.At(i).ScopeSpans()
+		for j := 0; j < scopespans.Len(); j++ {
+			scope := scopespans.At(j)
+			spans := scopespans.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				spandata := spans.At(k)
+				transformedadxtrace := mapToAdxTrace(resource.Resource(), scope.Scope(), spandata, e.logger)
+				adxtracejsonbytes, err := jsoniter.Marshal(transformedadxtrace)
+				if err != nil {
+					e.logger.Error("Error performing serialization of data.", zap.Error(err))
+				}
+				spanBuffer = append(spanBuffer, string(adxtracejsonbytes))
+			}
+		}
+	}
+	// Takes care of the residual data.
+	if len(spanBuffer) != 0 {
+		if err := e.ingestData(spanBuffer); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (adp *adxDataProducer) Close(context.Context) error {
```

Comment:
This looks like the only public API.
Is it needed by someone other than us (i.e. not internal)?
If so, please document

Reply:
The Close function will be used to override the exporter's default shutdown behavior. 
can be found in factory.go
`exporterhelper.WithShutdown(adp.Close)`